crates/ruff_linter/src/rules/ruff/rules/sequence_sorting.rs

```diff
@@ -0,0 +1,878 @@
+        //    up to the start of the first item in `__all__`/`__slots__`/etc.
+        //  - Postlude == the region of source code from the end of the last
+        //    item in `__all__`/`__slots__`/etc. up to and including the closing
+        //    parenthesis.
+        //
+        // For example:
+        //
+        // ```python
+        // __all__ = [  # comment0
+        //   # comment1
+        //   "first item",
+        //   "last item"  # comment2
+        //   # comment3
+        // ]  # comment4
+        // ```
+        //
+        // - The prelude in the above example is the source code region
+        //   starting just before the opening `[` and ending just after `# comment0`.
+        //   `comment0` here counts as part of the prelude because it is on
+        //   the same line as the opening paren, and because we haven't encountered
+        //   any elements of `__all__` yet, but `comment1` counts as part of the first item,
+        //   as it's on its own line, and all comments on their own line are grouped
+        //   with the next element below them to make "items",
+        //   (an "item" being a region of source code that all moves as one unit
+        //   when `__all__` is sorted).
+        // - The postlude in the above example is the source code region starting
+        //   just after `# comment2` and ending just after the closing paren.
+        //   `# comment2` is part of the last item, as it's an inline comment on the
+        //   same line as an element, but `# comment3` becomes part of the postlude
+        //   because there are no items below it. `# comment4` is not part of the
+        //   postlude: it's outside of the source-code range considered by this rule,
+        //   and should therefore be untouched.
+        //
+        let newline = stylist.line_ending().as_str();
+        let start_offset = self.start();
+        let leading_indent = leading_indentation(locator.full_line(start_offset));
+        let item_indent = format!("{}{}", leading_indent, stylist.indentation().as_str());
+
+        let prelude =
+            multiline_string_sequence_prelude(first_item_start, newline, start_offset, locator);
+        let postlude = multiline_string_sequence_postlude(
+            last_item_end,
+            newline,
+            leading_indent,
+            &item_indent,
+            self.end(),
+            locator,
+        );
+
+        self.items
+            .sort_by(|a, b| sorting_style.compare(&a.value, &b.value));
+        let joined_items = join_multiline_string_sequence_items(
+            &self.items,
+            locator,
+            &item_indent,
+            newline,
+            self.ends_with_trailing_comma,
+        );
+
+        format!("{prelude}{joined_items}{postlude}")
+    }
+}
+
+impl Ranged for MultilineStringSequenceValue {
+    fn range(&self) -> TextRange {
+        self.range
+    }
+}
+
+/// Collect data on each line of a multiline string sequence.
```

Comment:
How much of the below is new code vs. moved from the isort rule?

Reply:
I think everything below this point is entirely unchanged, _except_ that a few things have been renamed and a few comments tweaked (e.g. `collect_dunder_all_lines()` has been renamed to `collect_string_sequence_lines()`, etc.)